src/functional/ffront/foast_to_itir.py

```diff
@@ -157,11 +211,34 @@ def visit_Call(self, node: foast.Call, **kwargs) -> itir.FunCall:
         if TypeInfo(node.func.type).is_field_type:
             return self._visit_shift(node, **kwargs)
         elif node.func.id in FUN_BUILTIN_NAMES:
-            return self._visit_reduce(node, **kwargs)
-        return self._lift_lambda(node)(
+            visitor = getattr(self, f"_visit_{node.func.id}")
+            return visitor(node, **kwargs)
+        elif node.func.id in TYPE_BUILTIN_NAMES:
+            return self._visit_type_constr(node, **kwargs)
+        return self._lift_if_field(node)(
             im.call_(self.visit(node.func, **kwargs))(*self.visit(node.args, **kwargs))
         )
 
+    def _visit_neighbor_sum(self, node: foast.Call, **kwargs) -> itir.FunCall:
+        return self._visit_reduce(node, **kwargs)
+
+    def _visit_type_constr(self, node: foast.Call, **kwargs) -> itir.Literal:
+        if isinstance(node.args[0], foast.Constant):
+            target_type = fbuiltins.BUILTINS[node.func.id]
+            source_type = {**fbuiltins.BUILTINS, "string": str}[
+                node.args[0].dtype.kind.name.lower()
+            ]
+            if target_type is bool and source_type is not bool:
+                return im.literal_(str(bool(source_type(node.args[0].value))), node.func.id)
+            return im.literal_(node.args[0].value, node.func.id)
+        raise FieldOperatorLoweringError(f"Encountered a type cast, which is not supported: {node}")
+
+    def visit_Constant(self, node: foast.Constant, **kwargs) -> itir.Literal:
+        if isinstance(node.type, ct.ScalarType) and not node.type.shape:
+            typename = node.type.kind.name.lower()
```

Comment:
That should only happen to ints in the PAST lowering, because these are supposed to be ints. ITIR already has proper types. (Only backends might not respect them).

Reply:
There was a bug that did turn `int()` calls back into Python `int`s. A fix for it will be in the next commit.